src/gpg.rs

```diff
@@ -48,10 +50,13 @@ impl Gpg for LiveGpg {
             }
     }
 
-    fn par_receive_keys(key_server: &str, fingerprints: &[String]) -> Result<(), Box<dyn Error>> {
-        trace!("Receiving keys for fingerprints in PARALLEL");
+    fn par_receive_keys(key_server: &str, fingerprints: &HashSet<String>) -> Result<(), Box<dyn Error>> {
+        debug!("Receiving keys for fingerprints in PARALLEL");
+
         let _r : Vec<_> = fingerprints.par_iter().map(|fp| {
-            match Self::receive_keys(&key_server, &[fp.to_string()]){
+            let mut fingerprint_set = HashSet::new();
+            fingerprint_set.insert(fp.to_string());
+            match Self::receive_keys(&key_server, &fingerprint_set){
```

Comment:
If there were a `receive_key` variant then this code would be greatly simplified.